api/v1alpha1/packagebundle.go

```diff
@@ -26,22 +26,58 @@ func (config *PackageBundle) ExpectedKind() string {
 	return PackageBundleKind
 }
 
-func (config *PackageBundle) FindSource(pkgName, pkgVersion string) (retSource PackageOCISource, err error) {
+func (config *PackageBundle) FindPackage(pkgName string) (retPkg BundlePackage, err error) {
 	for _, pkg := range config.Spec.Packages {
 		if strings.EqualFold(pkg.Name, pkgName) {
-			source := pkg.Source
-			for _, packageVersion := range source.Versions {
-				// We do not sort before getting `latest` because there will be only a single packageVersion per release in normal cases. For edge cases which may require multiple
-				// versions, the order in the file will be ordered according to what we want `latest` to point to
-				if packageVersion.Name == pkgVersion || packageVersion.Digest == pkgVersion || pkgVersion == Latest {
-					retSource = PackageOCISource{Registry: source.Registry, Repository: source.Repository, Digest: packageVersion.Digest, Version: packageVersion.Name}
-					return retSource, nil
-				}
-			}
+			return pkg, nil
+		}
+	}
+	return retPkg, fmt.Errorf("package not found in bundle (%s): %s", config.ObjectMeta.Name, pkgName)
```

Comment:
I don't see a definition for retPkg. Am I missing something here? And should we be returning nil instead here?

Reply:
Named returns. See the method definition.